app.py

```diff
@@ -1,27 +1,114 @@
-from flask import Flask, render_template, request, redirect, url_for
-from view_model import ViewModel
+# Import dependnecies
+from flask import ( 
+    Flask, 
+    render_template, 
+    request, 
+    redirect, 
+    url_for, 
+    )
+from flask_login import (
+    LoginManager,
+    current_user,
+    login_required,
+    login_user,
+    logout_user,
+    )
 import logging
+
+# Appliction imports
+from view_model import ViewModel
 from mongo_db.item_store import ItemStore
+from mongo_db.user_store import UserStore
+from github_oauth import GithubOauthProvider
+from user.user import User
+from user.user_access import user_write_access, user_admin_access
+from user.user_role import UserRole
 
-def create_app(items = ItemStore()):
+def create_app(item_store = ItemStore(), user_store = UserStore()):
     app = Flask(__name__)
     logging.basicConfig(level=logging.INFO)
-    item_store = items
+
+    oauth_provider = GithubOauthProvider()
+
+    login_manager = LoginManager()
+    login_manager.init_app(app)
+
+    app.secret_key = oauth_provider.get_client_secret()
+    
+    @login_manager.unauthorized_handler
+    def unauthenticated():
+        # Authenticate with GitHub
+        return redirect(oauth_provider.get_authenticate_uri())
+    
+    @login_manager.user_loader
+    def load_user(user_id):
+        user = user_store.get_user(user_id)
+        return user
+
+    @app.route('/login/callback')
+    def login_callback():                                                            # pylint: disable=unused-variable
+        # Get authorization code from github 
+        code = request.args.get("code")
+
+        # Use the authorization code to get user info
+        userinfo = oauth_provider.get_user_info(
+            authorization_response=request.url,
+            redirect_url=request.base_url,
+            code=code)
+
+        user = User.from_json(userinfo)
+        user_store.add_user_if_missing(user)
+
+        # Begin user session
+        login_user(user)
+
+        return redirect("/")
+
+    @app.route("/logout")
+    @login_required
+    def logout():
+        logout_user()
+        return redirect("/")
+
+    @app.route("/admin")
+    @login_required
+    @user_admin_access
+    def admin():
+        users =  user_store.get_users()
+        return render_template('admin.html', users=users, login_disabled=is_login_disabled(),  current_user=current_user)
```

Comment:
It doesn't look like this login_disabled is used in admin.html

Reply:
removed